### PR TITLE
add new_item_label property

### PR DESCRIPTION
**Pull request by deluge**
I added a new property called "new_item_label" to the formset mixin. The reason why I added this, is, that we can access it in the template to show the correct label when it is initialized.

### CODE REVIEW

**Comment by stephrdev** [comment]
Wouldn't it be a good idea to have a `formset_context` property instead which would allow the user to add even more flags/values in a consistent interface?

**Comment by deluge** [comment]
Oh sure, I totally agree with you. I´ll update it.

**Comment by deluge** [comment]
I changed the code as discussed that we have a method "get_template_context" to add more custom data and a cached property "template_context" to get access to this data in templates.

File: cruditor/forms.py [hunk 1]
```diff
@@ -20,6 +20,13 @@ class CruditorFormsetMixin(object):
     """
     js_formset_options = None
 
+    @cached_property
```
Comment (stephrdev) [inline cruditor/forms.py:23]: cached_property is only available for Python 3.8+. We should leverage the `django.utils.functional.cached_property` instead.

File: cruditor/forms.py [hunk 2]
```diff
@@ -20,6 +20,13 @@ class CruditorFormsetMixin(object):
...
+    def template_context(self):
+        return self.get_template_context()
```
Comment (stephrdev) [inline cruditor/forms.py:25]: Some docus might be helpful.

File: tests/test_forms.py [hunk 3]
```diff
@@ -29,6 +29,13 @@ class JsOptionsDummyFormset(DummyFormset):
         'add-title': 'New one'
     }
 
+class TemplateContextDummyFormset(DummyFormset):
+    def get_template_context(self):
+        context = super().get_template_context()
+        context.update({'foo': 'bar'})
+
```
Comment (stephrdev) [inline tests/test_forms.py:36]:
```suggestion
```

**Review by stephrdev — CHANGES_REQUESTED** [review]

**Pull request merged**